database/mongodb.go

```diff
@@ -0,0 +1,36 @@
+package database
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func NewMongoDBClient(uri string, name string) (*mongo.Database, error) {
+
+	clientOption := options.Client().ApplyURI(uri)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	//connect to mongoDB
+	client, err := mongo.Connect(ctx, clientOption)
+	if err != nil {
+		return nil, err
+	}
+	//check the connection
+	err = client.Ping(ctx, readpref.Primary())
+	if err != nil {
+		return nil, err
+	}
+	log.Println("connected to monogodb!")
+
+	//get the database
+	mongoDB := client.Database(name)
+
+	return mongoDB, nil
```

Comment:
Since you're assigning the connection to the global connection variable...there's no need to return it again. Just return error only. So, update your function to only return error. 

If you are returning the connection, a user might end up reassigning the returned connection to another variable which means we end up with two references to the same connection pool.